src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -215,11 +218,14 @@ public function getQueryBuilder()
      *
      * @param string $sql SQL
      *
-     * @return \PDOStatement
+     * @return \PDOStatement|false
```

Comment:
Can we do |null instead please? The false part also due to PHP legacy is such an antipattern, and it wouldnt allow for future return types added as easily as just `?object`

Reply:
Isn't that a BC break because this emulates `PDO::query()`?

Reply:
0.next can do bc breaks no? If that means cleaning up the API, thats probably a good thing.
But if this is too low level, then never mind.

Reply:
I wouldn't bother. Cleaning up your own API :+1:, cleaning up PHP's :grimacing:.

Reply:
I agree with @Roy-Orbison here that I think it would be good to follow PHP's antipattern here just so that it's consistent with how one would expect `PDO::query()` to work (and similar comment to `PDO::execute()`.